rotkehlchen/data/importer.py

```diff
@@ -1014,3 +1015,169 @@ def import_shapeshift_trades_csv(self, filepath: Path) -> Tuple[bool, str]:
                 except KeyError as e:
                     return False, str(e)
         return True, ''
+
+    def _consume_uphold_transaction(self, csv_row: Dict[str, Any]) -> None:
+        """
+        Consume the file containing both trades and transactions from uphold.
+        """
+        timestamp = deserialize_timestamp_from_date(
+            date=csv_row['Date'],
+            formatstr='%a %b %d %Y %H:%M:%S %Z%z',
+            location='uphold',
+        )
+        destination = csv_row['Destination']
+        destination_asset = symbol_to_asset_or_token(csv_row['Destination Currency'])
+        destination_amount = deserialize_asset_amount(csv_row['Destination Amount'])
+        origin = csv_row['Origin']
+        origin_asset = symbol_to_asset_or_token(csv_row['Origin Currency'])
```

Comment:
This is a risky move. What it does is search the DB for an asset whose symbol matches. If there is none or if there is more than one it raises `UnknownAsset`. This is why for most exchanges we have mappings of known asset symbols to assets so that even if more assets are added in the DB with the same symbol (happens very very often unfortunately) there is no confusion in the long run.

Check this function as an example: https://github.com/rotki/rotki/blob/b9660102272522cbba7e438c05500149e64dfe98/rotkehlchen/assets/converters.py#L727-L750

Reply:
pending discussion, but I will use a converter to get the asset